packages/sdk/src/types/runtime/assets/videoStream.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Asset, AssetLike, AssetManager } from '.';
+import readPath from '../../../utils/readPath';
+import { InternalAsset } from '../../internal/asset';
+import { Patchable } from '../../patchable';
+
+export interface VideoStreamLike {
+	videoSourceType: VideoSourceType;
+	uri: string;
+	duration: number;
+}
+
+/** Type describing how to interpret the video Source URI. */
+export enum VideoSourceType {
```

Comment:
>VideoSourceType [](start = 12, length = 15)

why do we need to know the video type?

Reply:
Because the URI is not always a raw URL, and because it's clearer that we are doing special solution based on the type. For VideoSourceType=YouTube you can use URI `1nCqRmx3Dnw`, rather than `[https://www.youtube.com/watch?v=1nCqRmx3Dnw`](https://www.youtube.com/watch?v=1nCqRmx3Dnw`). And the type indicates that the player applies custom logic for choosing ideal stream for each user rather than one-bandwidth-fits-all


---
In reply to: [301160693](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/372#discussion_r301160693) [](ancestors = 301160693)

Reply:
I don't think we need to know or care about this on the SDK side.

---
In reply to: [301347617](https://github.com/microsoft/mixed-reality-extension-sdk/pull/372#discussion_r301347617) [](ancestors = 301347617,301160693)

Reply:
Seems like we're leaking an Altspace implementation detail.

---
In reply to: [301350379](https://github.com/microsoft/mixed-reality-extension-sdk/pull/372#discussion_r301350379) [](ancestors = 301350379,301347617,301160693)

Reply:
I don't think we should expose this. The client can encode it in the url.

---
In reply to: [301350449](https://github.com/microsoft/mixed-reality-extension-sdk/pull/372#discussion_r301350449) [](ancestors = 301350449,301350379,301347617,301160693)